### PR TITLE
Verify config and compile messages regex

### PR DESCRIPTION
**Pull request by loverend**
Add a check to verify the config. At the moment we are only using dicts
and strings in the config, so only added checks for those. If in future
we add any lists then we will need to update the config checker.

Complile the regex for each OS and store the dict ready to be checked
against. We will check the error title and if it matches then we will
use the regex, this should cut down on regex checks required.

### CODE REVIEW

File: napalm_logs/base.py [hunk 1]
```diff
@@ -130,6 +132,77 @@ def _load_config(self, path):
...
+                self._verify_config_key(key, value, valid, config, dev_os, key_path)
+
+    def _verify_config(self):
+        '''
+        Verify that the config is correct
+        '''
+        if not self.config_dict:
+            self._raise_config_exception('No config found')
+
+        valid_opts = {
```
Comment (mirceaulinic) [inline napalm_logs/base.py]: Can we move this as global?

The we won't need to pass it to `self._verify_config_dict`

File: napalm_logs/listener.py [hunk 2]
```diff
@@ -31,15 +31,15 @@ def start(self):
         '''
         Listen to messages and queue them.
         '''
+        buffer_size = 1024
```
Comment (mirceaulinic) [inline napalm_logs/listener.py]: Same, perhaps we should move this as a global and renamed to `BUFFER_SIZE` to be more obvious.

File: napalm_logs/base.py (hunk 1 above)
Comment (loverend) [inline napalm_logs/base.py]: I have made the variable global as it makes sense for it to be global, but I still have to pass it to `self._verify_config_dict` the first time it is called. It is a recursive function and each time the valid config for the specific section is passed.

File: napalm_logs/listener.py (hunk 2 above)
Comment (loverend) [inline napalm_logs/listener.py]: Done

**Pull request merged**